Issue/10635 material comments

khaykov opened the pull request:
Fixes #10635

This PR moves Comments List, Detail and Editor screens to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons.

Design changes are reviewed before the code review is requested.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on the "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

@mattmiklic Color of PENDING status was different in List and Details, so I changed one used in the list back to yellow.

![comparison_comments](https://user-images.githubusercontent.com/728822/71392930-e8ebc180-25be-11ea-8a11-2a0e0f5c2671.png)

To test:
Navigate to Comments management screen and confirm it's visual and functional integrity in both Light and Dark themes. The theme can be changed from App Settings.

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mattmiklic commented:
Looking good here. Two things I'm thinking about:

- The Spam/Trash/Edit icons and text on the comment detail screen are looking a little bit too high-contrast. Could we make those use the "medium emphasis" variant of the On Surface color?
- Could we remove the horizontal line on the Edit Comment screen? It doesn't seem necessary since there's no equivalent on the "Reply to post" comment screen.

khaykov commented:
@mattmiklic Sure, I updated the icons and label color. I removed the "medium emphasis" when secondary color is used (for like and approve actions) since they look really washed out.

![comparison_comments](https://user-images.githubusercontent.com/728822/71850410-163d7280-3089-11ea-9f06-ebae334e73f4.png)

The bottom line on Edit Comment magically disappeared in recent version :)

jkmassel commented:
I'm moving this PR to 14.1 since we're code freezing today.

develric commented on WordPress/src/main/java/org/wordpress/android/ui/CollapseFullScreenDialogFragment.java:
```diff
@@ -257,15 +263,16 @@ private void initBuilderArguments() {
-                onCollapseClicked();
-            }
-        });
+        toolbar.setNavigationOnClickListener(view1 -> onCollapseClicked());
```
I would consider to not use the numbering in the variable name (something like a simple `v` maybe?). Same in `CommentDetailFragment`. (I think the `view1` naming is the one created by AS to avoid the clash with the function parameter but I would consider to modify it as well, let me know wdyt 😊)

khaykov replied:
Good catch! Yes, AS creates those sometimes, even though it's inside a function. Weird.

develric commented on WordPress/src/main/java/org/wordpress/android/ui/comments/EditCommentActivity.java:
```diff
@@ -309,22 +309,15 @@ private void cancelEditCommentConfirmation() {
-                        // just close the dialog
-                    }
+                (dialog1, whichButton) -> {
+                    // just close the dialog
```
Same as in a previous comment, maybe I would use `dialog` (instead of `dialog1`)

develric commented on WordPress/src/main/res/values/dimens.xml:
```diff
@@ -434,7 +434,6 @@
     <dimen name="domain_suggestions_intro_header_bottom_padding">8dp</dimen>
 
-    <dimen name="keyline_first">16dp</dimen>
     <dimen name="keyline_second">72dp</dimen>
```
Not directly related to this PR, so no action required, but just to share this thought 😊; it sounded strange that we are not using anywhere the `keyline_first`. Probably we ended up using other `16dp` constant instead of this explicit one.

develric commented on WordPress/src/main/res/layout/comment_edit_activity.xml:
```diff
@@ -1,48 +1,68 @@
                 android:layout_width="match_parent"
-                android:layout_height="match_parent"
-                android:gravity="top"
-                android:hint="@string/hint_comment_content"
```
I'm pretty sure it is wanted but I prefer to cross check with you. I saw from one of your comment that we wanted to remove the horizontal line, but it seems we are also losing the hint, like in the following image:

![image](https://user-images.githubusercontent.com/47797566/72810425-d0260880-3c5d-11ea-949e-48b6a229016c.png)

khaykov replied:
Hints in `TextInputLayout` look a bit different - there is a label that goes to the top when you start typing and it doesn't really match the look. Considering that you will always open edit comment screen with some sort of comment (and there is an error if a comment is empty) I decided that we can get rid of it.

develric replied:
Thanks for the explanation Klym 😊!

khaykov commented:
Thanks for taking a look @develric ! I fixed the variable names :)
